cleanlab/classification.py

```diff
@@ -689,23 +697,38 @@ def find_label_issues(
 
         if self.verbose:
             print("Using predicted probabilities to identify label issues ...")
-        self.label_issues_mask = filter.find_label_issues(
+        label_issues_mask = filter.find_label_issues(
             labels,
             pred_probs,
             **self.find_label_issues_kwargs,
         )
         label_quality_scores = get_label_quality_scores(
             labels, pred_probs, **self.label_quality_scores_kwargs
         )
-        self.label_issues_df = pd.DataFrame(
-            {"is_label_issue": self.label_issues_mask, "label_quality": label_quality_scores}
+        label_issues_df = pd.DataFrame(
+            {"is_label_issue": label_issues_mask, "label_quality": label_quality_scores}
         )
-        self.label_issues_df["given_label"] = labels
-        self.label_issues_df["predicted_label"] = pred_probs.argmax(axis=1)
         if self.verbose:
-            print(f"Identified {np.sum(self.label_issues_mask)} examples with label issues.")
-
-        return self.label_issues_df
+            print(f"Identified {np.sum(label_issues_mask)} examples with label issues.")
+
+        predicted_labels = pred_probs.argmax(axis=1)
+        compressed_type = None
+        if self.num_classes < np.iinfo(np.dtype("int16")).max:
```

Comment:
This logic should go in utils. its fairly universal and we should use it to reduce space of ints everywhere.

Reply:
where else should it be used?

Reply:
moved to utils function. This function is now also being applied to `labels` as I benchmarked memory and realized the labels are being duplicated in order to create the dataframe, so may as well compress them.